cores/esp8266/umm_malloc/umm_malloc.cpp

```diff
@@ -1610,9 +1764,12 @@ static void *_umm_realloc( void *ptr, size_t size ) {
     /* And don't forget to adjust the pointer to the new block location! */
 
     ptr    = (void *)&UMM_DATA(c);
+
+    /* Now resume critical section... */
+    UMM_CRITICAL_ENTRY(id_realloc);
   }
```

Comment:
what if the above memmove gets interrupted by an ISR that happens to write to the mem chunk being moved?

Reply:
As a shared resource that memory allocation may become corrupted. 

An ISR that chooses to do `realloc` or `free` has to implement some locking method to ensure exclusive access when doing these operations, Similar to what a multithreaded APP would have to do with a shared resource.

Reply:
An option that could be added, would be to define two new critical section macros something like:
```cpp
 #ifdef UMM_LIGHTWEIGHT_CPU
 #define UMM_CRITICAL_SUSPEND(tag) UMM_CRITICAL_EXIT(tag)
 #define UMM_CRITICAL_RESUME(tag) UMM_CRITICAL_ENTRY(tag)
 #else
 #define UMM_CRITICAL_SUSPEND(tag)
 #define UMM_CRITICAL_RESUME(tag)
 #endif
```
These would replace the newer `UMM_CRITICAL_EXIT` and `UMM_CRITICAL_ENTRY` that were added to enable IRQs for memcpys. This would allow for easy backward compatibility.

Reply:
Hmm, a worse scenario would be the foreground APP is writing to memory and the writing process is interrupted and the ISR realloc(s) space and the space the foreground is using goes away or is then owned by a different ISR due to two IRQs back to back each using realloc/malloc.  When the call returns the foreground code is writing on someone else's memory. I think this gets back to the foreground and ISRs have to coordinate shared resources through some form of lock. -- Oh how I miss `interlockedExchange()`.